### ISSUE TITLE
restoreRevision with fake fields error 500 still exists

### ISSUE THREAD

**Issue by dontfreakout** [Laravel-Backpack/CRUD#791]
Hello, 

the problem #353 still exists in latest CRUD and Laravel. 
I tried to trace the problem:
Eloquent updates model in **performUpdate()** and there it finds what to update using **getDirty()** method: 

_(in L5.3: `Illuminate\Database\Eloquent\Model` class, 
in L5.4: `Illuminate\Database\Eloquent\Concerns\HasAttributes` Trait)_

```php
    /**
     * Get the attributes that have been changed since last sync.
     *
     * @return array
     */
    public function getDirty()
    {
        $dirty = [];

        foreach ($this->attributes as $key => $value) {
            // Set to update values in attributes which are not in $originals array
            if (! array_key_exists($key, $this->original)) {
                $dirty[$key] = $value;
            } elseif ($value !== $this->original[$key] &&
                                 ! $this->originalIsNumericallyEquivalent($key)) {
                $dirty[$key] = $value;
            }
        }

        return $dirty;
    }
```
It selects all attributes which are modified or are not in `$originals`, therefore all fake fields.
I tested it in Tinker and if you get the model with `withFakes()` and you try to update something, it fails with `Column not found` error as @kichetof posted at #353 . If you try the same model without fakes, it updates.

### My solution

Simplest solution for me was to modify **getEntry()** method in `Backpack\CRUD\PanelTraits\Read` trait to accept second param `$withFakes` to optionally not include fake fields. And **restoreRevision()** in `Backpack\CRUD\PanelTraits\ViewsAndRestoresRevisions` trait to call `getEntry($id, $withFakes = false)`

**Backpack\CRUD\PanelTraits\Read:**
```php
    /**
     * Find and retrieve an entry in the database or fail.
     *
     * @param  [int] The id of the row in the db to fetch.
     *
     * @return [Eloquent Collection] The row in the db.
     */
    public function getEntry($id, $withFakes = true)
    {
        $this->entry = $this->model->findOrFail($id);
        if ($withFakes === true) {
            $this->entry = $this->entry->withFakes();
        }

        return $this->entry;
    }
```

**Backpack\CRUD\PanelTraits\ViewsAndRestoresRevisions:**
```php
    /**
     * Restore a single revision.
     *
     * @param  [int] $id         The ID of the source CRUD Model instance to update
     * @param  [int] $revisionId The ID of the revision to use for the update
     */
    public function restoreRevision($id, $revisionId)
    {
       // Do not include fake fields
        $entry = $this->getEntry($id, false);
        $revision = Revision::findOrFail($revisionId);

        // Update the revisioned field with the old value
        $entry->update([$revision->key => $revision->old_value]);

        // Reload the entry so we have the latest revisions
        $entry = $this->getEntry($id);
    }
```

Thank you for this awesome admin package :+1:  
If you need anything from me, feel free to ask :)

Bests
Martin

_PHP 5.6.30
Laravel 5.3, 5.4
Mysql 10.1.24-MariaDB
Backpack CRUD 3.2.13_

**Comment by lloy0076** [comment]
The solution won't work if `$this->entry` already exists __and__ has the fakes, though.

**Comment by lloy0076** [comment]
If this is still an issue that we need to resolve (as opposed to one that will simply clutter up the issues), please re-open. Preferably with a PR :)

**Comment by luukScheur** [comment]
This issue is still present in the latest build. The fix above works with me. Commenting on @lloy0076, the latest Read.php [now](https://github.com/Laravel-Backpack/CRUD/blob/master/src/PanelTraits/Read.php#L60) checks if `$this->entry` already exists, so inserting above changes inside this if statement should do it right?

**Comment by tabacitu** [comment]
Oooooh!!! I get it now! The 500 error showed up _when trying to restore a change to the ```extras``` attribute_. Ok, I could replicate this and can confirm it's a bug. I've [applied the fix here](https://github.com/Laravel-Backpack/CRUD/commit/cc915cb65c8a93ad8e5627eb19fa46c8dd647487) - through an extra method instead of adding a second parameter, and have merged it into the ```3.5``` branch, which will be released later this week.

Thanks a lot for a comprehensive bug report @dontfreakout . I'm sorry I didn't get to this sooner.

Cheers!
